FinSearchUnified/BusinessLogic/ExportService.php

```diff
@@ -0,0 +1,208 @@
+            ->orderBy('articles.id')
+            ->setParameter('active', true);
+
+        return $countQuery->getQuery()->getScalarResult()[0][1];
+    }
+
+    public function fetchAllProducts($start, $count)
+    {
+        $articlesQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('articles')
+            ->where('articles.active = :active')
+            ->orderBy('articles.id')
+            ->setParameter('active', true);
+
+        if ($count > 0) {
+            $articlesQuery->setMaxResults($count)->setFirstResult($start);
+        }
+
+        return $articlesQuery->getQuery()->execute();
+    }
+
+    public function fetchProductById($productId)
+    {
+         $articlesQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('articles')
+            ->where('articles.id = :productId')
+            ->orWhere('articles.supplier = :productId')
+            ->setParameter('productId', $productId);
+
+        $shopwareArticles = $articlesQuery->getQuery()->execute();
+
+        if (count($shopwareArticles) === 0) {
+            $this->errors['general'][] = 'No article found with given ID';
+            return null;
+        }
+
+        return $shopwareArticles;
+    }
+
+    public function generateFindologicProducts($shopwareArticles, $logErrors = false)
+    {
+        $findologicArticles = [];
+
+        try {
+            /** @var Article $article */
+            foreach ($shopwareArticles as $article) {
+                $findologicArticle = self::getFindologicArticle($article, $logErrors);
+
+                if ($findologicArticle) {
+                    $findologicArticles[] = $findologicArticle;
+                }
+            }
+        } catch (Exception $e) {
+            die($e->getMessage());
+        }
+
+        return $findologicArticles;
+    }
+
+    /**
+     * @param Article $shopwareArticle
+     * @param bool $logErrors
+     * @return null|Item
+     * @throws Exception
+     */
+    public function getFindologicArticle($shopwareArticle, $logErrors)
+    {
+        $articleId = $shopwareArticle->getId();
+        $inactiveCatCount = 0;
+        $totalCatCount = 0;
+
+        /** @var Category $category */
+        foreach ($shopwareArticle->getCategories() as $category) {
+            if (!$category->isChildOf($this->baseCategory)) {
+                continue;
+            }
+
+            if (!$category->getActive()) {
+                $inactiveCatCount++;
+            }
+
+            $totalCatCount++;
+        }
+
+        if (!$shopwareArticle->getActive()) {
+            if ($logErrors) {
+                $this->errors[$articleId][] = 'Product is not active';
+            }
+            return null;
+        }
+
+        if ($totalCatCount === $inactiveCatCount) {
+            if ($logErrors) {
+                $this->errors[$articleId][] = sprintf('All %d categories are inactive', $totalCatCount);
```

Comment:
I feel like the count of properties does not really matter. Knowing that all categories are inactive is sufficient information.